Refactor eos cli config gen

carlbuchmann opened the pull request:
<!-- All PR should follow this template to allow a clean and transparent review -->
<!-- Bot will manage tag, ownership and code review, you should not update these fields-->
## Change Summary

Refactor eos_cli_config_gen to leverage new filters to improve code readability.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)
<!-- If PR is linked to one or more issues, please list issues below -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

## Component(s) name

eos_cli_config_gen

__EOS templates__

- [x] aaa-accounting.j2 (CB)
  - [X] Review (CH)
  - [ ] Approved (CH)
- [x] aaa-authentication.j2 (CB)
  - [x] Review (CH)
  - [ ] Approved (CH)
- [ ] aaa-authorization.j2 (XR)
  - [x] Review (CH)
  - [ ] Approved (CH)
- [ ] aaa-root.j2(XR)
  - [x] Review (CH)
  - [ ] Approved (CH)
- [ ] aaa-server-groups.j2 (XR)
  - [x] Review (CH)
  - [ ] Approved (CH)
- [ ] access-lists.j2 (XR)
  - [x] Review (CH)
  - [ ] Approved (CH)
- [x] aliases.j2 (XR)
  - [x] Review (CH)
  - [x] Approved (CH)
- [x] arp.j2 (XR)
  - [x] Review (CH)
  - [x] Approved (CH)
- [x] banners.j2 (XR)
  - [x] Review (CH)
  - [x] Approved (CH)
- [ ] clock-timezone.j2 (XR)
  - [x] Review (CH)
  - [ ] Approved (CH)
- [ ] clock-timezone.j2 (XR)
- [ ] community-lists.j2 (XR)
- [ ] custom-templates.j2 (XR)
- [ ] daemon-templates.j2 (XR)
- [ ] daemon-terminattr.j2 (XR)
- [ ] dns-domain.j2 (XR)
- [ ] domain-list.j2 (XR)
- [ ] enable-password.j2 (XR)
- [ ] errdisable.j2 (XR)
- [ ] ethernet-interfaces.j2 (XR)
- [ ] event-handler.j2 (XR)
- [ ] event-monitor.j2 (XR)
- [ ] hardware-counters.j2 (XR)
- [ ] hardware.j2 (XR)
- [ ] interface-defaults.j2 (XR)
- [ ] ip-dhcp-relay.j2 (XR)
- [ ] ip-extended-communities.j2 (XR)
- [ ] ip-igmp-snooping.j2 (XR) 
- [ ] ip-routing.j2 (XR)
- [ ] ip-tacacs-source-interfaces.j2 to mac-security.j2 (XR)
- [x] router-bgp.j2 (CB)
  - [x] Review (CH)
  - [ ] Approved (CH)
- [ ] spanning-tree.j2 (OG)
- [ ] standard-access-lists.j2 (OG)
- [ ] static-routes.j2 (OG)
- [ ] tacacs-servers.j2 (OG)
- [x] tcam-profile.j2 (OG)
- [x] terminal-settings.j2 (OG)
- [x] unsupported-transceiver.j2(OG)
- [x] virtual-router-mac-address.j2 (OG)
  - [x] Review (CH)
  - [x] Approved (CH)
- [x] vlan-internal-order.j2 (OG)
  - [x] Review (CH)
  - [ ] Approved (CH)
- [x] vlans.j2 (OG)
  - [x] Review (CH)
  - [ ] Approved (CH)
- [x] virtual-source-nat.j2 (OG)
  - [x] Review (CH)
  - [ ] Approved (CH)
- [x] vlan-interfaces.j2 (TG)
  - [x] Review (CH)
  - [ ] Approved (CH)
- [x] vmtracer-sessions.j2 (TG)
  - [x] Review (CH)
  - [ ] Approved (CH)
- [x] vrf-instances.j2 (TG)
  - [x] Review (CH)
  - [ ] Approved (CH)
- [x] vxlan-interface.j2 (TG)
  - [x] Review (CH)
  - [ ] Approved (CH)




__Documentation templates__

...

## Proposed changes

- update to leverage arista.avd.defined

## How to test

- Run molecule scenario and ensure no functional changes

carlbuchmann committed:
refactor arista.avd.defined

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:
```diff
@@ -1,129 +1,129 @@
    {{ bgp_default }}
 {%     endfor %}
-{%     if router_bgp.peer_groups is defined and router_bgp.peer_groups is not none%}
+{%     if router_bgp.peer_groups is arista.avd.defined %}
```
the natural_sort now contains this check :)

carlbuchmann committed:
refactor for loop

carlbuchmann committed:
refactor long lines

carlbuchmann committed:
refactor aaa new coding standards

carlbuchmann committed:
refactor aaa authentication new coding standards

ClausHolbechArista committed:
Missing maximum_routes from rebase conflict fix

ClausHolbechArista committed:
Update CI Artifacts

ClausHolbechArista reviewed (approved):
LGTM!

I have rebased and updated molecule

titom73 committed:
Refactor eos_cli_config vlan-interfaces

titom73 committed:
Update CI artifacts

titom73 committed:
Refactor eos_cli_config vxlan-interface

titom73 committed:
Refactor eos_cli_config vrf-instances

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-accounting.j2:
```diff
@@ -1,16 +1,22 @@
-{%     if aaa_accounting.exec.default is defined and aaa_accounting.exec.default is not none %}
-{%        if aaa_accounting.exec.default.group is defined and aaa_accounting.exec.default.group is not none %}
+{%     if aaa_accounting.exec.default is arista.avd.defined %}
+{%        if aaa_accounting.exec.default.group is arista.avd.defined %}
```
No check that `aaa_accounting.exec.default.group.type` is defined

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-accounting.j2:
```diff
@@ -1,16 +1,22 @@
 !
-{%     if aaa_accounting.exec.default is defined and aaa_accounting.exec.default is not none %}
-{%        if aaa_accounting.exec.default.group is defined and aaa_accounting.exec.default.group is not none %}
+{%     if aaa_accounting.exec.default is arista.avd.defined %}
```
No check that `aaa_accounting.exec` is defined

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-authentication.j2:
```diff
@@ -1,24 +1,24 @@
+{% if aaa_authentication is arista.avd.defined %}
 !
-{%     if aaa_authentication.login.default is defined and aaa_authentication.login.default is not none %}
+{%     if aaa_authentication.login.default is arista.avd.defined %}
```
No check for `aaa_authentication.login` is defined

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-authentication.j2:
```diff
@@ -1,24 +1,24 @@
 aaa authentication login serial-console {{ aaa_authentication.login.serial_console }}
 {%     endif %}
-{%     if aaa_authentication.dot1x.default is defined and aaa_authentication.dot1x.default is not none %}
+{%     if aaa_authentication.dot1x.default is arista.avd.defined %}
```
No check for `aaa_authentication.dot1x` is defined

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:
```diff
@@ -1,458 +1,470 @@
-{%       for peer_group in router_bgp.peer_groups | arista.avd.natural_sort %}
-{%          if router_bgp.peer_groups[peer_group].bgp_listen_range_prefix is defined and router_bgp.peer_groups[peer_group].bgp_listen_range_prefix is not none %}
+{%     for peer_group in router_bgp.peer_groups | arista.avd.natural_sort %}
+{%         if router_bgp.peer_groups[peer_group].bgp_listen_range_prefix is arista.avd.defined %}
```
No check if `router_bgp.peer_groups[peer_group].peer_filter` is defined

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:
```diff
@@ -1,458 +1,470 @@
       rd {{ router_bgp.vlans[vlan].rd }}
-{%            if router_bgp.vlans[vlan].route_targets.both is defined and router_bgp.vlans[vlan].route_targets.both is not none %}
-{%                for route_target in router_bgp.vlans[vlan].route_targets.both %}
+{%         if router_bgp.vlans[vlan].route_targets.both is arista.avd.defined %}
```
no check if `router_bgp.vlans[vlan].route_targets` is defined

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:
```diff
@@ -1,458 +1,470 @@
+{%     for vlan in router_bgp.vlans | arista.avd.natural_sort %}
    !
    vlan {{ vlan }}
       rd {{ router_bgp.vlans[vlan].rd }}
```
no check if `router_bgp.vlans[vlan].rd` is defined

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:
```diff
@@ -1,458 +1,470 @@
+{%     for vlan_aware_bundle in router_bgp.vlan_aware_bundles | arista.avd.natural_sort %}
    !
    vlan-aware-bundle {{ vlan_aware_bundle }}
       rd {{ router_bgp.vlan_aware_bundles[vlan_aware_bundle].rd }}
```
No check if `router_bgp.vlan_aware_bundles[vlan_aware_bundle].rd` is defined

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:
```diff
@@ -1,458 +1,470 @@
+{%             if router_bgp.address_family_evpn.peer_groups[peer_group].route_map_out is arista.avd.defined %}
       neighbor {{ peer_group }} route-map {{ router_bgp.address_family_evpn.peer_groups[peer_group].route_map_out }} out
 {%             endif %}
 {%             if router_bgp.address_family_evpn.peer_groups[peer_group].activate == true %}
```
No check if `router_bgp.address_family_evpn.peer_groups[peer_group].activate` is defined

titom73 committed:
Refactor eos_cli_config vmtracer-sessions

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:
```diff
@@ -1,458 +1,470 @@
-{%            for peer_group in router_bgp.address_family_rtc.peer_groups | arista.avd.natural_sort %}
-{%               if router_bgp.address_family_rtc.peer_groups[peer_group].activate == true %}
+{%         for peer_group in router_bgp.address_family_rtc.peer_groups | arista.avd.natural_sort %}
+{%             if router_bgp.address_family_rtc.peer_groups[peer_group].activate == true %}
```
No check if `router_bgp.address_family_rtc.peer_groups[peer_group].activate` is defined

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:
```diff
@@ -1,458 +1,470 @@
+{%             if router_bgp.address_family_ipv4.peer_groups[peer_group].prefix_list_out is arista.avd.defined %}
       neighbor {{ peer_group }} prefix-list {{ router_bgp.address_family_ipv4.peer_groups[peer_group].prefix_list_out }} out
 {%             endif %}
 {%             if router_bgp.address_family_ipv4.peer_groups[peer_group].activate == true %}
```
No check if `router_bgp.address_family_ipv4.peer_groups[peer_group].activate` is defined

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:
```diff
@@ -1,458 +1,470 @@
+{%                 endif %}
+      {{ neighbor_default_originate_cli }}
 {%             endif %}
 {%             if router_bgp.address_family_ipv4.neighbors[neighbor].activate == true %}
```
No check if `router_bgp.address_family_ipv4.neighbors[neighbor].activate` is defined

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:
```diff
@@ -1,458 +1,470 @@
+{%             if router_bgp.address_family_ipv4_multicast.peer_groups[peer_group].route_map_out is arista.avd.defined %}
       neighbor {{ peer_group }} route-map {{ router_bgp.address_family_ipv4_multicast.peer_groups[peer_group].route_map_out }} out
 {%             endif %}
 {%             if router_bgp.address_family_ipv4_multicast.peer_groups[peer_group].activate == true %}
```
No check if `router_bgp.address_family_ipv4_multicast.peer_groups[peer_group].activate` is defined

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:
```diff
@@ -1,458 +1,470 @@
+{%             if router_bgp.address_family_ipv4_multicast.neighbors[neighbor].route_map_out is arista.avd.defined %}
       neighbor {{ neighbor }} route-map {{ router_bgp.address_family_ipv4_multicast.neighbors[neighbor].route_map_out }} out
 {%             endif %}
 {%             if router_bgp.address_family_ipv4_multicast.neighbors[neighbor].activate == true %}
```
No check if `router_bgp.address_family_ipv4_multicast.neighbors[neighbor].activate` is defined

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:
```diff
@@ -1,458 +1,470 @@
+{%             if router_bgp.address_family_ipv6.peer_groups[peer_group].route_map_out is arista.avd.defined %}
       neighbor {{ peer_group }} route-map {{ router_bgp.address_family_ipv6.peer_groups[peer_group].route_map_out }} out
 {%             endif %}
 {%             if router_bgp.address_family_ipv6.peer_groups[peer_group].activate == true %}
```
No check if `router_bgp.address_family_ipv6.peer_groups[peer_group].activate` is defined

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:
```diff
@@ -1,458 +1,470 @@
+{%             if router_bgp.address_family_ipv6.neighbors[neighbor].route_map_out is arista.avd.defined %}
       neighbor {{ neighbor }} route-map {{ router_bgp.address_family_ipv6.neighbors[neighbor].route_map_out }} out
 {%             endif %}
 {%             if router_bgp.address_family_ipv6.neighbors[neighbor].activate == true %}
```
No check if `router_bgp.address_family_ipv6.neighbors[neighbor].activate` is defined

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:
```diff
@@ -1,458 +1,470 @@
+{%     for vrf in router_bgp.vrfs | arista.avd.natural_sort %}
    !
    vrf {{ vrf }}
       rd {{ router_bgp.vrfs[vrf].rd }}
```
No check if `router_bgp.vrfs[vrf].rd` is defined

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:
```diff
@@ -1,458 +1,470 @@
-{%             if router_bgp.vrfs[vrf].route_targets.import is defined and router_bgp.vrfs[vrf].route_targets.import is not none %}
-{%                for address_family in router_bgp.vrfs[vrf].route_targets.import %}
-{%                      for route_target in router_bgp.vrfs[vrf].route_targets.import[address_family] %}
+{%         if router_bgp.vrfs[vrf].route_targets.import is arista.avd.defined %}
```
No check if `router_bgp.vrfs[vrf].route_targets` is defined

ClausHolbechArista reviewed (commented):
Not sure if we want to check for all of them, if they are required values.

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2:
```diff
@@ -1,155 +1,169 @@
 {# eos - VLAN Interfaces #}
-{% if vlan_interfaces is defined and vlan_interfaces is not none %}
+{% if vlan_interfaces is arista.avd.defined %}
```
This check is included in natural_sort, so we no longer need to have this if outside the for-loops, as long as we use natural_sort.

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2:
```diff
@@ -1,155 +1,169 @@
+{%         elif vlan_interfaces[vlan_interface].shutdown is arista.avd.defined(false) %}
    no shutdown
 {%         endif %}
 {%         if vlan_interfaces[vlan_interface].mtu is defined and vlan_interfaces[vlan_interface].mtu != 1500 %}
```
The new style is to configure MTU no matter what the value is. Defaults could have been changed to something else than 1500. Also use arista.avd.defined filter.

onurgashi committed:
Refactor vlan-internal-order.j2

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2:
```diff
@@ -1,155 +1,169 @@
    arp aging timeout {{ vlan_interfaces[vlan_interface].arp_aging_timeout }}
 {%         endif %}
-{%             if vlan_interfaces[vlan_interface].ip_proxy_arp is defined and vlan_interfaces[vlan_interface].ip_proxy_arp == true %}
+{%             if vlan_interfaces[vlan_interface].ip_proxy_arp is arista.avd.defined(true) %}
```
Wrong indentation

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2:
```diff
@@ -1,155 +1,169 @@
-{%             if vlan_interfaces[vlan_interface].ip_proxy_arp is defined and vlan_interfaces[vlan_interface].ip_proxy_arp == true %}
+{%             if vlan_interfaces[vlan_interface].ip_proxy_arp is arista.avd.defined(true) %}
    ip proxy-arp
 {%             endif %}
```
Wrong indentation

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2:
```diff
@@ -1,155 +1,169 @@
    ip address virtual {{ vlan_interfaces[vlan_interface].ip_address_virtual }}
 {%         endif %}
-{%         if vlan_interfaces[vlan_interface].ip_helpers is defined and vlan_interfaces[vlan_interface].ip_helpers is not none %}
+{%         if vlan_interfaces[vlan_interface].ip_helpers is arista.avd.defined %}
```
This check is included in natural_sort, so we no longer need to have this if outside the for-loops, as long as we use natural_sort.

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2:
```diff
@@ -1,155 +1,169 @@
-{%         if vlan_interfaces[vlan_interface].multicast.ipv4.source_route_export.enabled is defined and vlan_interfaces[vlan_interface].multicast.ipv4.source_route_export.enabled == true %}
-   multicast ipv4 source route export{% if vlan_interfaces[vlan_interface].multicast.ipv4.source_route_export.administrative_distance is defined and vlan_interfaces[vlan_interface].multicast.ipv4.source_route_export.administrative_distance is not none %} {{ vlan_interfaces[vlan_interface].multicast.ipv4.source_route_export.administrative_distance }}{% endif %}
-
+{%         if vlan_interfaces[vlan_interface].multicast.ipv4.source_route_export.enabled is arista.avd.defined(true) %}
```
Missing check for `vlan_interfaces[vlan_interface].multicast` and `vlan_interfaces[vlan_interface].multicast.ipv4` and `vlan_interfaces[vlan_interface].multicast.ipv4.source_route_export`

onurgashi committed:
Refactor virtual-router-mac-address.j2

onurgashi committed:
Refactor virtual-source-nat.j2

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2:
```diff
@@ -1,155 +1,169 @@
    ip ospf authentication-key 7 {{ vlan_interfaces[vlan_interface].ospf_authentication_key }}
 {%         endif %}
-{%         if vlan_interfaces[vlan_interface].ospf_message_digest_keys is defined and vlan_interfaces[vlan_interface].ospf_message_digest_keys is not none %}
+{%         if vlan_interfaces[vlan_interface].ospf_message_digest_keys is arista.avd.defined %}
```
This check is included in natural_sort, so we no longer need to have this if outside the for-loops, as long as we use natural_sort.

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2:
```diff
@@ -1,155 +1,169 @@
-{%         if vlan_interfaces[vlan_interface].ospf_message_digest_keys is defined and vlan_interfaces[vlan_interface].ospf_message_digest_keys is not none %}
+{%         if vlan_interfaces[vlan_interface].ospf_message_digest_keys is arista.avd.defined %}
 {%           for ospf_message_digest_key in vlan_interfaces[vlan_interface].ospf_message_digest_keys | arista.avd.natural_sort %}
    ip ospf message-digest-key {{ ospf_message_digest_key }} {{ vlan_interfaces[vlan_interface].ospf_message_digest_keys[ospf_message_digest_key].hash_algorithm }} 7 {{ vlan_interfaces[vlan_interface].ospf_message_digest_keys[ospf_message_digest_key].key }}
```
no check for `hash_algorithm` and `key`

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2:
```diff
@@ -1,155 +1,169 @@
-{%         if vlan_interfaces[vlan_interface].pim is defined and vlan_interfaces[vlan_interface].pim is not none %}
-{%            if vlan_interfaces[vlan_interface].pim.ipv4.sparse_mode is defined and vlan_interfaces[vlan_interface].pim.ipv4.sparse_mode == true %}
+{%         if vlan_interfaces[vlan_interface].pim is arista.avd.defined %}
+{%            if vlan_interfaces[vlan_interface].pim.ipv4.sparse_mode is arista.avd.defined(true) %}
```
no check for `vlan_interfaces[vlan_interface].pim.ipv4`

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2:
```diff
@@ -1,155 +1,169 @@
-{%             if vlan_interfaces[vlan_interface].vrrp.priority is defined and vlan_interfaces[vlan_interface].vrrp.priority is not none and vlan_interfaces[vlan_interface].vrrp.priority > 1 %}
+{%         if vlan_interfaces[vlan_interface].vrrp is arista.avd.defined %}
+{%           if vlan_interfaces[vlan_interface].vrrp.virtual_router is arista.avd.defined %}
+{%             if vlan_interfaces[vlan_interface].vrrp.priority is arista.avd.defined and vlan_interfaces[vlan_interface].vrrp.priority | int > 1 %}
```
just configure supplied priority without evaluating > 1.

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/virtual-source-nat.j2:
```diff
@@ -1,5 +1,5 @@
 {# eos - virtual source-nat - introduced in 4.21 #}
-{% if virtual_source_nat_vrfs is defined and virtual_source_nat_vrfs is not none %}
+{% if virtual_source_nat_vrfs is arista.avd.defined %}
```
This check is now included in natural_sort filter, so we no longer need to test before the for-loop, as long as it uses natural_sort.

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/virtual-source-nat.j2:
```diff
@@ -1,5 +1,5 @@
+{% if virtual_source_nat_vrfs is arista.avd.defined %}
 !
 {%     for vrf in virtual_source_nat_vrfs | arista.avd.natural_sort %}
 ip address virtual source-nat vrf {{ vrf }} address {{ virtual_source_nat_vrfs[vrf].ip_address }}
```
No check if `virtual_source_nat_vrfs[vrf].ip_address` is defined

xaviramon committed:
Refactor different eos files

onurgashi committed:
Refactor vlans.j2

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-internal-order.j2:
```diff
@@ -1,5 +1,5 @@
-{% if vlan_internal_order is defined and vlan_internal_order is not none %}
+{% if vlan_internal_order is arista.avd.defined %}
 !
 vlan internal order {{ vlan_internal_order.allocation }} range {{ vlan_internal_order.range.beginning }} {{ vlan_internal_order.range.ending }}
```
Missing checks for `vlan_internal_order.allocation` and `vlan_internal_order.range.beginning` and `vlan_internal_order.range.ending`

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlans.j2:
```diff
@@ -4,10 +4,10 @@
 !
```
Remove check in line 2 since it is part of natural_sort

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlans.j2:
```diff
@@ -4,10 +4,10 @@
 !
 vlan {{ vlan }}
    name {{ vlans[vlan].name }}
```
missing check for `.name`

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlans.j2:
```diff
@@ -4,10 +4,10 @@
    state {{ vlans[vlan].state }}
 {%         endif %}
-{%         if vlans[vlan].trunk_groups is defined %}
+{%         if vlans[vlan].trunk_groups is arista.avd.defined %}
```
This check is now part of `natural_sort` so you don't need it before the for-loop

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vmtracer-sessions.j2:
```diff
@@ -1,22 +1,22 @@
 {# eos - vmtracer sessions #}
-{% if vmtracer_sessions is defined and vmtracer_sessions is not none %}
+{% if vmtracer_sessions is arista.avd.defined %}
```
This check is now part of natural_sort so you don't need it before the for-loop

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vrf-instances.j2:
```diff
@@ -1,9 +1,9 @@
 {# eos - VRFs #}
-{% if vrfs is defined and vrfs is not none %}
+{% if vrfs is arista.avd.defined %}
```
This check is now part of natural_sort so you don't need it before the for-loop

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vxlan-interface.j2:
```diff
@@ -1,21 +1,20 @@
 interface Vxlan1
    vxlan source-interface {{ vxlan_tunnel_interface.Vxlan1.source_interface }}
-{%     if vxlan_tunnel_interface.Vxlan1.virtual_router is defined %}
+{%     if vxlan_tunnel_interface.Vxlan1.virtual_router is arista.avd.defined %}
```
Missing check for `vxlan_tunnel_interface.Vxlan1`

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vxlan-interface.j2:
```diff
@@ -1,21 +1,20 @@
 {%     endif %}
-   vxlan udp-port {{ vxlan_tunnel_interface.Vxlan1.vxlan_udp_port }}
+   vxlan udp-port {{ vxlan_tunnel_interface.Vxlan1.vxlan_udp_port | arista.avd.default('4789') }}
 {%     if vxlan_tunnel_interface.Vxlan1.vxlan_vni_mappings.vlans is iterable %}
```
This check is now part of natural_sort so you don't need it before the for-loop.
But missing check for `vxlan_tunnel_interface.Vxlan1.vxlan_vni_mappings`

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vxlan-interface.j2:
```diff
@@ -1,21 +1,20 @@
+   vxlan udp-port {{ vxlan_tunnel_interface.Vxlan1.vxlan_udp_port | arista.avd.default('4789') }}
 {%     if vxlan_tunnel_interface.Vxlan1.vxlan_vni_mappings.vlans is iterable %}
 {%          for vlan in vxlan_tunnel_interface.Vxlan1.vxlan_vni_mappings.vlans | arista.avd.natural_sort %}
    vxlan vlan {{ vlan }} vni {{ vxlan_tunnel_interface.Vxlan1.vxlan_vni_mappings.vlans[vlan].vni }}
```
missing check for `vxlan_tunnel_interface.Vxlan1.vxlan_vni_mappings.vlans[vlan].vni`

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vxlan-interface.j2:
```diff
@@ -1,21 +1,20 @@
-{% if vxlan_tunnel_interface.Vxlan1.vxlan_vni_mappings.vrfs is defined and
-vxlan_tunnel_interface.Vxlan1.vxlan_vni_mappings.vrfs is not none %}
-{%     for vrf in vxlan_tunnel_interface.Vxlan1.vxlan_vni_mappings.vrfs | arista.avd.natural_sort %}
+{%     if vxlan_tunnel_interface.Vxlan1.vxlan_vni_mappings.vrfs is arista.avd.defined %}
```
This check is now part of natural_sort so you don't need it before the for-loop.
But missing check for vxlan_tunnel_interface.Vxlan1.vxlan_vni_mappings (best to add it above before the vlans and nest both vlans and vrf inside the if)

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vxlan-interface.j2:
```diff
@@ -1,21 +1,20 @@
-{%     for vrf in vxlan_tunnel_interface.Vxlan1.vxlan_vni_mappings.vrfs | arista.avd.natural_sort %}
+{%     if vxlan_tunnel_interface.Vxlan1.vxlan_vni_mappings.vrfs is arista.avd.defined %}
+{%         for vrf in vxlan_tunnel_interface.Vxlan1.vxlan_vni_mappings.vrfs | arista.avd.natural_sort %}
    vxlan vrf {{ vrf }} vni {{ vxlan_tunnel_interface.Vxlan1.vxlan_vni_mappings.vrfs[vrf].vni }}
```
missing check for `vxlan_tunnel_interface.Vxlan1.vxlan_vni_mappings.vrfs[vrf].vni`

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vxlan-interface.j2:
```diff
@@ -1,21 +1,20 @@
-{% endif %}
+{%         endfor %}
+{%     endif %}
 {% endif %}
```
Missing newline at the end of the file

xaviramon committed:
Merge branch 'refactor-eos-cli-config-gen' of github.com:carlbuchmann/ansible-avd into refactor-eos-cli-config-gen

onurgashi committed:
Refac/rename + add mole unsupported-transceiver.j2

onurgashi committed:
Refactor terminal-settings.j2

onurgashi committed:
Refactor tcam-profile.j2

onurgashi committed:
Fix whitespace tcam-profile.j2

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-authorization.j2:
```diff
@@ -1,15 +1,15 @@
-{% if aaa_authorization is defined and aaa_authorization is not none %}
-{%     if aaa_authorization.exec.default is defined and aaa_authorization.exec.default is not none %}
+{% if aaa_authorization is arista.avd.defined  %}
+{%     if aaa_authorization.exec.default is arista.avd.defined %}
```
missing check for `aaa_authorization.exec`

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-authorization.j2:
```diff
@@ -1,15 +1,15 @@
 no aaa authorization config-commands
 {%     endif %}
-{%     if aaa_authorization.commands.all_default is defined and aaa_authorization.commands.all_default is not none %}
+{%     if aaa_authorization.commands.all_default is arista.avd.defined %}
```
missing check for `aaa_authorization.commands`

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-root.j2:
```diff
@@ -1,4 +1,5 @@
+{# eos - aaa root #}
+{% if aaa_root is arista.avd.defined %}
 !
 {%     if aaa_root.secret is defined %}
```
change to `is arista.avd.defined`

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-root.j2:
```diff
@@ -1,4 +1,5 @@
+{% if aaa_root is arista.avd.defined %}
 !
 {%     if aaa_root.secret is defined %}
 aaa root secret sha512 {{ aaa_root.secret.sha512_password }}
```
missing check for variable `aaa_root.secret.sha512_password`

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/access-lists.j2:
```diff
@@ -1,5 +1,5 @@
+{% if access_lists is arista.avd.defined %}
 !
 {%     for access_list in access_lists | arista.avd.natural_sort %}
 ip access-list {{ access_list }}
```
missing check for `access_lists[access_list].sequence_numbers[sequence].action` in line 7

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-server-groups.j2:
```diff
@@ -1,11 +1,10 @@
+   server {{ server['server'] }}{% if server['vrf'] is arista.avd.defined %} vrf {{ server['vrf'] }}{% endif %}
+{%       endfor %}
+{%    endfor %}
 {% endif %}
```
missing newline at the end of the file

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-server-groups.j2:
```diff
@@ -1,11 +1,10 @@
-{%          endfor %}
-{%      endfor %}
+{%       for server in aaa_server_group.servers %}
+   server {{ server['server'] }}{% if server['vrf'] is arista.avd.defined %} vrf {{ server['vrf'] }}{% endif %}
```
this may produce bad output with missing newlines at the end of the cli line, so please refactor to the new style where we built the cli first (check recording of last meeting or ping me)

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-server-groups.j2:
```diff
@@ -1,11 +1,10 @@
 !
-{%     for aaa_server_group in aaa_server_groups %}
+{%    for aaa_server_group in aaa_server_groups %}
 aaa group server {{ aaa_server_group['type'] }} {{ aaa_server_group['name'] }}
```
missing checks for variable `aaa_server_group.type` and `aaa_server_group.name`
Also refactor to use `.` instead of `[ ]`

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/clock-timezone.j2:
```diff
@@ -1,4 +1,5 @@
+{% if clock is arista.avd.defined %}
 !
 clock timezone {{ clock.timezone }}
 {% endif %}
```
missing newline at the end of the file

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/clock-timezone.j2:
```diff
@@ -1,4 +1,5 @@
+{# eos - Clock timezone#}
+{% if clock is arista.avd.defined %}
 !
 clock timezone {{ clock.timezone }}
```
missing check for `clock.timezone`

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/community-lists.j2:
```diff
@@ -1,5 +1,5 @@
+{% if community_lists is arista.avd.defined %}
 !
 {%     for community_list in community_lists | arista.avd.natural_sort %}
 ip community-list {{ community_list }} {{ community_lists[community_list].action }}
```
missing check for `community_lists[community_list].action`

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/daemon-terminattr.j2:
```diff
@@ -1,4 +1,5 @@
+{% if daemon_terminattr is arista.avd.defined %}
 !
 {# Create list of non CVaaS server #}
 {% set cvp_on_prem = namespace() %}
```
missing indentation and most of this fine need refactoring to make it readable (set the cli variable and add options)

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/dns-domain.j2:
```diff
@@ -1,4 +1,4 @@
-{% if dns_domain is defined and dns_domain is not none %}
+{% if dns_domain is arista.avd.defined %}
 dns domain {{ dns_domain }}
 {% endif %}
```
missing newline and the end of the file

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/domain-list.j2:
```diff
@@ -1,5 +1,5 @@
 {# eos - domain-list #}
-{% if domain_list is defined and domain_list is not none %}
+{% if domain_list is arista.avd.defined %}
 {%    for domain in domain_list | sort %}
```
change to `| arista.avd.natural_sort` which also contains the check in line 2, so you can remove that `if`

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/domain-lookup.j2:
```diff
@@ -1,7 +1,7 @@
+{% if ip_domain_lookup is arista.avd.defined %}
 {%    for source in ip_domain_lookup.source_interfaces | arista.avd.natural_sort %}
-{%              if ip_domain_lookup.source_interfaces[source].vrf is defined and ip_domain_lookup.source_interfaces[source].vrf is not none %}
+{%              if ip_domain_lookup.source_interfaces[source].vrf is arista.avd.defined %}
```
indentation is wrong

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/errdisable.j2:
```diff
@@ -1,5 +1,5 @@
 {# eos - Errdisable #}
-{% if errdisable is defined and errdisable is not none %}
+{% if errdisable is arista.avd.defined %}
 {%     if errdisable.recovery is defined %}
```
use arista.avd.defined

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/errdisable.j2:
```diff
@@ -1,5 +1,5 @@
-{% if errdisable is defined and errdisable is not none %}
+{% if errdisable is arista.avd.defined %}
 {%     if errdisable.recovery is defined %}
 {%         if errdisable.detect.causes is defined %}
```
This check is included in natural_sort so you don't need it before the for-loop, but we are missing a check for `errdisable.detect` so you can just change it to that. Please use arista.avd.defined as well.

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/errdisable.j2:
```diff
@@ -1,5 +1,5 @@
+{% if errdisable is arista.avd.defined %}
 {%     if errdisable.recovery is defined %}
 {%         if errdisable.detect.causes is defined %}
 {%             for cause in errdisable.detect.causes | arista.avd.natural_sort %}
```
refactor line 7 to `if cause in ['acl', 'arp-inspection' .....` so we don't need to have all these ifs.
Same for other areas of this file.

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/errdisable.j2:
```diff
@@ -59,7 +59,7 @@ errdisable recovery cause {{ cause }}
 {%                 endif %}
 {%             endfor %}
 {%         endif %}
-{%         if errdisable.recovery.interval is defined and errdisable.recovery.interval is not none  %}
```
missing check for `errdisable.recovery`. Add it above the `causes` section and nest both `causes` and `interval` under that `if`

titom73 committed:
Refactor vlan-interfaces as per peer-review

- Remove unecessary if arista.avd.defined
- Be explicit for MTU settings
- Be explicit for VRRP priority
- Fix indentation

titom73 committed:
Update vlan-interfaces artifacts

carlbuchmann commented:
Closing moving to individual PRs
